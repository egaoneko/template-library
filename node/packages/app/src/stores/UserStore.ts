import { IUser } from '@my-app/core/lib/interfaces/user';
import { makeAutoObservable } from 'mobx';
import { useMemo } from 'react';

class UserStore {
  public user: IUser | null = null;

  constructor(user: IUser | null) {
    makeAutoObservable(this);
    void this.clear();
    void this.hydrate(user);
  }

  public async hydrate(user: IUser | null): Promise<void> {
    await this.setUser(user);
  }

  private async clear(): Promise<void> {
    await this.setUser(null);
  }

  private async setUser(user: IUser | null): Promise<void> {
    this.user = user;
  }
}

export default UserStore;

let store: UserStore;

function initializeStore(user: IUser | null) {
  const _store = store ?? new UserStore(user);

  if (store?.user?.id !== _store.user?.id) {
    store = _store;
  }

  return _store;
}

export function useUserStore(user: IUser | null): UserStore {
  return useMemo(() => initializeStore(user), [user]);
}
