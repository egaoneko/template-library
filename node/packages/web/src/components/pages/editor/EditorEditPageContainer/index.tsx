import Head from '@components/atoms/layout/Head';
import React, { FC, useState } from 'react';
import EditorUpdateContentTemplate from './templates/EditorUpdateContentTemplate';
import { useRouter } from 'next/router';
import { BasePropsType } from '@my-app/core/lib/interfaces/common';
import BaseLayoutTemplate from '@components/templates/layout/BaseLayoutTemplate';
import { IArticle, UpdateArticleRequest } from '@my-app/core/lib/interfaces/article';
import ArticleAPI from '@api/article';
import { notifyError, notifySuccess } from '@utils/notifiy';
import { CONTEXT } from '@constants/common';

interface PropsType extends BasePropsType {
  article: IArticle;
}

const EditorEditPageContainer: FC<PropsType> = props => {
  const { article } = props;
  const router = useRouter();
  const [loading, setLoading] = useState<boolean>(false);

  async function onFinish(request: UpdateArticleRequest): Promise<void> {
    setLoading(true);
    return ArticleAPI.update(CONTEXT, article.slug, request)
      .then(article => {
        if (!article) {
          setLoading(false);
          return;
        }
        notifySuccess('Successfully updated!');
        router.push(`/article/${article.slug}`);
      })
      .catch(e => {
        setLoading(false);
        notifyError(e.response?.data?.message ?? e.message);
      });
  }

  return (
    <BaseLayoutTemplate pathname={props.pathname}>
      <Head title={'EDIT ARTICLE'} />
      <EditorUpdateContentTemplate loading={loading} article={article} onFinish={onFinish} />
    </BaseLayoutTemplate>
  );
};

export default EditorEditPageContainer;
