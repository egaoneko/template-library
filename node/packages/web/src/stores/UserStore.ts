import UserAPI from '@api/user';
import { makeAutoObservable } from 'mobx';
import { IS_SSR } from '@constants/common';
import { useMemo } from 'react';
import { IUser, LoginRequest, RegisterRequest, UpdateRequest } from '@interfaces/user';
import AuthAPI from '@api/auth';
import { notifyError, notifySuccess } from '@utils/notifiy';
import { removeToken, setToken } from '@utils/cookie';

export class UserStore {
  public lastUpdate: number = 0;
  public user: IUser | null = null;

  constructor(user: IUser | null) {
    makeAutoObservable(this);
    void this.clear();
    void this.hydrate(user);
  }

  public async hydrate(user: IUser | null): Promise<void> {
    await this.setUser(user);
  }

  public async register(request: RegisterRequest): Promise<boolean> {
    try {
      const user = await AuthAPI.register(request);
      notifySuccess('Successfully registered!');
      return !!user;
    } catch (e) {
      notifyError(e.response?.data?.message ?? e.message);
    }

    return false;
  }

  public async login(request: LoginRequest): Promise<IUser | null> {
    try {
      const user = await AuthAPI.login(request);
      this.setUser(user);

      if (this.user) {
        notifySuccess('Successfully login!');
      } else {
        notifyError('Fail to login');
      }
    } catch (e) {
      notifyError(e.response?.data?.message ?? e.message);
    }

    return this.user;
  }

  public async update(request: UpdateRequest): Promise<IUser | null> {
    try {
      const user = await UserAPI.update(request);
      this.setUser({
        ...this.user,
        ...user,
      });

      if (this.user) {
        notifySuccess('Successfully updated!');
      } else {
        notifyError('Fail to update');
      }
    } catch (e) {
      notifyError(e.response?.data?.message ?? e.message);
    }

    return this.user;
  }

  public async logout(): Promise<void> {
    try {
      await AuthAPI.logout();
      removeToken();
      await this.clear();
    } catch (e) {
      notifyError(e.message);
    }
  }

  private async clear(): Promise<void> {
    await this.setUser(null);
  }

  private async setUser(user: IUser | null): Promise<void> {
    this.user = user;

    if (!this.user || !this.user?.token) {
      return;
    }

    setToken(this.user.token, this.user.refreshToken);
  }
}

let store: UserStore;

function initializeStore(user: IUser | null) {
  const _store = store ?? new UserStore(user);

  if (IS_SSR) {
    return _store;
  }

  if (store?.user?.id !== _store.user?.id) {
    store = _store;
  }

  return _store;
}

export function useUserStore(user: IUser | null): UserStore {
  return useMemo(() => initializeStore(user), [user?.id]);
}
