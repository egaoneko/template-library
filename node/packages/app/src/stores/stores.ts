import { useContext } from 'react';
import { MobXProviderContext } from 'mobx-react';

import UserStore from './UserStore';

export interface Stores {
  userStore: UserStore;
}

export function useStores(): Stores {
  return useContext(MobXProviderContext) as Stores;
}
