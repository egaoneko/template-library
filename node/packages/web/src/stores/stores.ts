import { useContext } from 'react';
import { enableStaticRendering, MobXProviderContext } from 'mobx-react';
import { IS_SSR } from 'src/constants/common';
import { UserStore } from './UserStore';

enableStaticRendering(IS_SSR);

export interface Stores {
  userStore: UserStore;
}

export function useStores(): Stores {
  return useContext(MobXProviderContext) as Stores;
}
