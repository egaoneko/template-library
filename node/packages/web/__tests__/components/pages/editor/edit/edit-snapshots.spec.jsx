import React from 'react';
import renderer from 'react-test-renderer';
import { UserStore } from '../../../../../src/stores/UserStore';
import { Provider } from 'mobx-react';
import EditorEditPageContainer from '../../../../../src/components/pages/editor/EditorEditPageContainer';

describe('Editor New Snapshots', () => {
  let userStore;

  beforeEach(() => {
    userStore = new UserStore();
  });

  it('renders editor new', async () => {
    await userStore.hydrate(require('../../../../../__mocks__/user/user.json'));
    const tree = renderer
      .create(
        <Provider userStore={userStore}>
          <EditorEditPageContainer article={require('../../../../../__mocks__/article/article.json')} />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
