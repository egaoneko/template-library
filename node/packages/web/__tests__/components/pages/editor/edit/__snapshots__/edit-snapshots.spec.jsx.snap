// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor New Snapshots renders editor new 1`] = `
.c7 {
  width: 100%;
  position: absolute;
  bottom: 0px;
  margin-top: 3rem;
  --tw-bg-opacity: 1;
  background-color: rgba(229,231,235,var(--tw-bg-opacity));
  padding-left: 3.5rem;
  padding-right: 3.5rem;
  padding-top: 1rem;
  padding-bottom: 1rem;
  font-size: 0.75rem;
  line-height: 1rem;
  --tw-text-opacity: 1;
  color: rgba(156,163,175,var(--tw-text-opacity));
}

.c8 {
  color: var(--color-primary);
}

.c8:hover {
  color: var(--color-secondary);
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c5 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 9999px;
}

.c6 {
  border-radius: 9999px;
}

.c1 {
  width: 100%;
  margin-left: auto;
  margin-right: auto;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c2 {
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c3 {
  font-weight: 700;
  font-size: 1.5rem;
  line-height: 2rem;
  color: var(--color-primary);
  font-family: ui-sans-serif,system-ui,-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,"Helvetica Neue",Arial,"Noto Sans",sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol","Noto Color Emoji";
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  width: 100%;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

@media (min-width:640px) {
  .c1 {
    max-width: 640px;
  }
}

@media (min-width:768px) {
  .c1 {
    max-width: 768px;
  }
}

@media (min-width:1024px) {
  .c1 {
    max-width: 1024px;
  }
}

@media (min-width:1280px) {
  .c1 {
    max-width: 1280px;
  }
}

@media (min-width:1536px) {
  .c1 {
    max-width: 1536px;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <div
        className="c3"
        data-cy="header-logo-link"
      >
        <a
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          conduit
        </a>
      </div>
      <div
        className="text-base leading-6 text-gray-400 font-sans flex items-center"
      >
        <div
          className="py-2 ml-4 cursor-pointer first:ml-0 hover:text-gray-800 flex "
          data-cy="header-home-link"
        >
          <div
            className="flex items-center"
            href="/"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            <span
              className="mr-1"
            />
            <span>
              Home
            </span>
          </div>
        </div>
        <div
          className="py-2 ml-4 cursor-pointer first:ml-0 hover:text-gray-800 flex "
          data-cy="header-new-post-link"
        >
          <div
            className="flex items-center"
            href="/editor/new"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            <span
              className="mr-1"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 1024 1024"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 0 0 0-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 0 0 9.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
                />
              </svg>
            </span>
            <span>
              New Post
            </span>
          </div>
        </div>
        <div
          className="py-2 ml-4 cursor-pointer first:ml-0 hover:text-gray-800 flex "
          data-cy="header-settings-link"
        >
          <div
            className="flex items-center"
            href="/user/settings"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            <span
              className="mr-1"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 1024 1024"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M924.8 625.7l-65.5-56c3.1-19 4.7-38.4 4.7-57.8s-1.6-38.8-4.7-57.8l65.5-56a32.03 32.03 0 0 0 9.3-35.2l-.9-2.6a443.74 443.74 0 0 0-79.7-137.9l-1.8-2.1a32.12 32.12 0 0 0-35.1-9.5l-81.3 28.9c-30-24.6-63.5-44-99.7-57.6l-15.7-85a32.05 32.05 0 0 0-25.8-25.7l-2.7-.5c-52.1-9.4-106.9-9.4-159 0l-2.7.5a32.05 32.05 0 0 0-25.8 25.7l-15.8 85.4a351.86 351.86 0 0 0-99 57.4l-81.9-29.1a32 32 0 0 0-35.1 9.5l-1.8 2.1a446.02 446.02 0 0 0-79.7 137.9l-.9 2.6c-4.5 12.5-.8 26.5 9.3 35.2l66.3 56.6c-3.1 18.8-4.6 38-4.6 57.1 0 19.2 1.5 38.4 4.6 57.1L99 625.5a32.03 32.03 0 0 0-9.3 35.2l.9 2.6c18.1 50.4 44.9 96.9 79.7 137.9l1.8 2.1a32.12 32.12 0 0 0 35.1 9.5l81.9-29.1c29.8 24.5 63.1 43.9 99 57.4l15.8 85.4a32.05 32.05 0 0 0 25.8 25.7l2.7.5a449.4 449.4 0 0 0 159 0l2.7-.5a32.05 32.05 0 0 0 25.8-25.7l15.7-85a350 350 0 0 0 99.7-57.6l81.3 28.9a32 32 0 0 0 35.1-9.5l1.8-2.1c34.8-41.1 61.6-87.5 79.7-137.9l.9-2.6c4.5-12.3.8-26.3-9.3-35zM788.3 465.9c2.5 15.1 3.8 30.6 3.8 46.1s-1.3 31-3.8 46.1l-6.6 40.1 74.7 63.9a370.03 370.03 0 0 1-42.6 73.6L721 702.8l-31.4 25.8c-23.9 19.6-50.5 35-79.3 45.8l-38.1 14.3-17.9 97a377.5 377.5 0 0 1-85 0l-17.9-97.2-37.8-14.5c-28.5-10.8-55-26.2-78.7-45.7l-31.4-25.9-93.4 33.2c-17-22.9-31.2-47.6-42.6-73.6l75.5-64.5-6.5-40c-2.4-14.9-3.7-30.3-3.7-45.5 0-15.3 1.2-30.6 3.7-45.5l6.5-40-75.5-64.5c11.3-26.1 25.6-50.7 42.6-73.6l93.4 33.2 31.4-25.9c23.7-19.5 50.2-34.9 78.7-45.7l37.9-14.3 17.9-97.2c28.1-3.2 56.8-3.2 85 0l17.9 97 38.1 14.3c28.7 10.8 55.4 26.2 79.3 45.8l31.4 25.8 92.8-32.9c17 22.9 31.2 47.6 42.6 73.6L781.8 426l6.5 39.9zM512 326c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm79.2 255.2A111.6 111.6 0 0 1 512 614c-29.9 0-58-11.7-79.2-32.8A111.6 111.6 0 0 1 400 502c0-29.9 11.7-58 32.8-79.2C454 401.6 482.1 390 512 390c29.9 0 58 11.6 79.2 32.8A111.6 111.6 0 0 1 624 502c0 29.9-11.7 58-32.8 79.2z"
                />
              </svg>
            </span>
            <span>
              Settings
            </span>
          </div>
        </div>
        <div
          className="py-2 ml-4 cursor-pointer first:ml-0 hover:text-gray-800 flex "
          data-cy="header-profile-link"
        >
          <div
            className="flex items-center"
            href="/profile/Jacob"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            <span
              className="mr-1"
            />
            <span>
              <div
                className="flex items-center gap-1"
              >
                <div
                  className="c4"
                >
                  <div
                    className="c5 w-5 h-5"
                  >
                    <span
                      style={
                        Object {
                          "background": "none",
                          "border": 0,
                          "bottom": 0,
                          "boxSizing": "border-box",
                          "display": "block",
                          "height": "initial",
                          "left": 0,
                          "margin": 0,
                          "opacity": 1,
                          "overflow": "hidden",
                          "padding": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                          "width": "initial",
                        }
                      }
                    >
                      <img
                        alt="avatar"
                        className="c6 rounded-full"
                        data-nimg="fill"
                        decoding="async"
                        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                        style={
                          Object {
                            "border": "none",
                            "bottom": 0,
                            "boxSizing": "border-box",
                            "display": "block",
                            "height": 0,
                            "left": 0,
                            "margin": "auto",
                            "maxHeight": "100%",
                            "maxWidth": "100%",
                            "minHeight": "100%",
                            "minWidth": "100%",
                            "objectFit": undefined,
                            "objectPosition": undefined,
                            "padding": 0,
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                            "width": 0,
                          }
                        }
                      />
                      <noscript>
                        <img
                          alt="avatar"
                          className="c6 rounded-full"
                          data-nimg="fill"
                          decoding="async"
                          loading="lazy"
                          sizes="100vw"
                          src="/_next/image?url=https%3A%2F%2Fi.stack.imgur.com%2FxHWG8.jpg&w=3840&q=75"
                          srcSet="/_next/image?url=https%3A%2F%2Fi.stack.imgur.com%2FxHWG8.jpg&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fi.stack.imgur.com%2FxHWG8.jpg&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fi.stack.imgur.com%2FxHWG8.jpg&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fi.stack.imgur.com%2FxHWG8.jpg&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fi.stack.imgur.com%2FxHWG8.jpg&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fi.stack.imgur.com%2FxHWG8.jpg&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fi.stack.imgur.com%2FxHWG8.jpg&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fi.stack.imgur.com%2FxHWG8.jpg&w=3840&q=75 3840w"
                          style={
                            Object {
                              "border": "none",
                              "bottom": 0,
                              "boxSizing": "border-box",
                              "display": "block",
                              "height": 0,
                              "left": 0,
                              "margin": "auto",
                              "maxHeight": "100%",
                              "maxWidth": "100%",
                              "minHeight": "100%",
                              "minWidth": "100%",
                              "objectFit": undefined,
                              "objectPosition": undefined,
                              "padding": 0,
                              "position": "absolute",
                              "right": 0,
                              "top": 0,
                              "width": 0,
                            }
                          }
                        />
                      </noscript>
                    </span>
                  </div>
                </div>
                Jacob
              </div>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="container mx-auto mt-8"
  >
    <div
      className="w-full max-w-3xl mx-auto"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="flex flex-col gap-4"
        >
          <div>
            <input
              className="text-xl font-bold px-6 py-3 block w-full rounded-md border-gray-300 shadow-sm focus:border-indigo-300 focus:ring focus:ring-indigo-200 focus:ring-opacity-50 disabled:opacity-50 disabled:cursor-not-allowed"
              data-cy="content-form-input-title"
              defaultValue="How to train your dragon"
              disabled={false}
              name="title"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Article Title"
              type="text"
            />
          </div>
          <div>
            <input
              className=" block w-full rounded-md border-gray-300 shadow-sm focus:border-indigo-300 focus:ring focus:ring-indigo-200 focus:ring-opacity-50 disabled:opacity-50 disabled:cursor-not-allowed"
              data-cy="content-form-input-description"
              defaultValue="Ever wonder how?"
              disabled={false}
              name="description"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="What's this article about?"
              type="text"
            />
          </div>
          <div>
            <textarea
              className=" mt-1 block w-full rounded-md border-gray-300 shadow-sm focus:border-indigo-300 focus:ring focus:ring-indigo-200 focus:ring-opacity-50"
              data-cy="content-form-input-body"
              defaultValue="It takes a **Jacobian**"
              disabled={false}
              name="body"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Write your article (in markdown)"
              rows={8}
            />
          </div>
          <button
            className="ml-auto px-6 py-3 hover:text-opacity-80 hover:bg-opacity-80 hover:border-opacity-80 font-bold border rounded focus:outline-none focus:shadow-outline disabled:opacity-50 disabled:cursor-not-allowed bg-primary border-primary text-white"
            data-cy="content-form-button-submit"
            disabled={false}
            type="submit"
          >
            Publish Article
          </button>
        </div>
      </form>
    </div>
  </div>
  <div
    className="c7"
  >
    <span
      className="c8 text-base mr-1 font-bold"
      data-cy="footer-logo-link"
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      conduit
    </span>
    © 2020. An interactive learning project from
    <span
      className="c8 ml-1"
      data-cy="footer-site-link"
      href="https://thinkster.io/"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      Thinkster
    </span>
    . Code licensed under MIT.
  </div>
</div>
`;
